src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -68,6 +68,7 @@ public int Take()
             }
 
             var index = _index;
+            var isLastBlock = block.Next == null;
```

Comment:
Nit: you should rename this variable to "wasLastBlock".  The prefix "is" implies something which is true / false at this very moment.  That is not the case here.  At best you can say that at a previous point in time this was the last block.